### ISSUE TITLE
[a11y design review]: Modernized Check-in Experience aka unified appointment check-in experience

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#57212]
## Auditor(s)

Sara Smith and Sarah Koomson

## Design Review
Based on the development of the prototype, some of the following may or may not be applicable. 
- [ ] [Color and contrast checks](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#use-of-color)
  - [ ] WCAG 2.1 AA
  - [ ] WCAG 2.1 AAA
  - [ ] [Who Can Use](https://www.whocanuse.com/?b=ecece5&c=000000&f=20&s=)
  - [ ] High Contrast Mode
- [ ] [Keyboard navigation](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#keyboard-nav)
  - [ ] Annotations for Tab order, reflow and reading order (if needed)
- [ ] Accessible Name Annotations
  - [ ] [Alt attributes for images](https://webaim.org/techniques/alttext/)
  - [ ] Text alternative for complex images such as charts graphs maps etc.
- [ ] Language (ease of reading)
- [ ] HTML Tags
  - [ ] [Page Title](https://accessibility.18f.gov/page-titles/)
  - [ ] [Heading order](https://accessibility.18f.gov/headings/)
   - [ ] Annotations (if needed)
  - [ ] Semantic Consideration for paragraphs ordered lists numbering
  - [ ] [Tables (heading rows)](https://design.va.gov/components/table)
- [ ] Size of clickable areas

### As Time Allows
- [ ] [Components](https://design.va.gov/components/) follow VA design system guidance.
- [ ] [Patterns](https://design.va.gov/patterns/) follow VA design system guidance.

## Prototype Link
**Taken from Ticket [Accessibility Specialist Research Support Request from Modernized Check-In Experience Team#56137](https://app.zenhub.com/workspaces/sitewide-content-accessibility-and-ia-63a1d63232beba0011a7833f/issues/gh/department-of-veterans-affairs/va.gov-team/56137)**
- Link to [product outline](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/checkin/product/README.md)
- Link to or attach user flows: [Sketch file](https://www.sketch.com/s/0e890de3-2530-4ee0-986e-cf0314334aec) and link to an [individual sketch artboard where the user flow begins](https://www.sketch.com/s/0e890de3-2530-4ee0-986e-cf0314334aec/a/bgwvQKZ)
- Link to prototype or mockups: CodePen prototype not available yet 
- Link to any epics/issues that may be helpful: [Collab cycle ticket](https://app.zenhub.com/workspaces/check-in-experience-61fc23a2cb8a14001132e102/issues/gh/department-of-veterans-affairs/va.gov-team/53488)
### Epic 

**Sitewide Content, Accessibility, and IA work for unified appointment check-in experience** #[55813](https://app.zenhub.com/workspaces/sitewide-content-accessibility-and-ia-63a1d63232beba0011a7833f/issues/gh/department-of-veterans-affairs/va.gov-team/55813)

### Tickets

1. CAIA Intake #[55686](https://app.zenhub.com/workspaces/sitewide-content-accessibility-and-ia-63a1d63232beba0011a7833f/issues/gh/department-of-veterans-affairs/va.gov-team/55686)
2. Content Work: #[55836](https://app.zenhub.com/workspaces/sitewide-content-accessibility-and-ia-63a1d63232beba0011a7833f/issues/gh/department-of-veterans-affairs/va.gov-team/55686)
3. Accessibility Research: #[56137](https://app.zenhub.com/workspaces/sitewide-content-accessibility-and-ia-63a1d63232beba0011a7833f/issues/gh/department-of-veterans-affairs/va.gov-team/56137)
4. This ticket a11y design review #[57212](https://app.zenhub.com/workspaces/sitewide-content-accessibility-and-ia-63a1d63232beba0011a7833f/issues/gh/department-of-veterans-affairs/va.gov-team/57212) 

### Non-Distinguishable Links
- **Sketch File Location:** [Appointment List - Pre-Visit Task](https://www.sketch.com/s/0e890de3-2530-4ee0-986e-cf0314334aec/a/bgwvQKZ#Inspect)
- **Details**: The "details" buttons on the page will need to have an ARIA label applied to make the page navigable. 
- **Rationale**: Screen readers will otherwise have to go line-by-line on the page to figure out the connections to the links; whereas, if the ARIA label is added they can navigate the page successfully. 
- **Image Reference**: [View an image reference to the non-distinguishable links issue in the comments](https://github.com/department-of-veterans-affairs/va.gov-team/issues/57212#issuecomment-1518209388). 

#### Bandaid solution
- NA

#### Recommended solution
- Check for the title or ARIA attributes to provide context or additional off-screen text. 
- **Example**: aria-label="Details about Monday, February 2, 8:00 a.m. Primary Care with Johnathan Scmidt">Details</a>

#### Headings
- **Sketch File Location:** [Appointment List - Pre-Visit Task](https://www.sketch.com/s/0e890de3-2530-4ee0-986e-cf0314334aec/a/bgwvQKZ#Inspect)
- **Details**: Headings should be in logical order and levels should not be skipped. 
   - **H1** Page title reading order jumps to **H4** Task Card “Save time by confirming your appointment before you arrive.” It should be an **H2**. 
   - The heading order goes from **H2** “February 2023” to **H3** “Need help,” at the bottom of the page; however, “Need Help” should be an **H2**. 
   - **Related**:_Also, see additional notes on headings in card section that will be added to this ticket._ 
- **Rationale**: As a screen reader user, users want to navigate the hierarchy of the page content using heading levels to save time and frustration.
- **Image Reference**: [View an image reference to the heading order issue in the comments](https://github.com/department-of-veterans-affairs/va.gov-team/issues/57212#issuecomment-1518209388). 
- **Additional Resource**: [WCAG 2.1 Section Headings](https://www.w3.org/WAI/WCAG21/Understanding/section-headings)

##### Bandaid solution
- The heading styles should be applied in the correct order. 

##### Recommended solution
- Use CSS to style the header visually if you need a different visual representation.

#### Card Semantic Structure

- **Sketch File Location:** [Appointment List - Pre-Visit Task](https://www.sketch.com/s/0e890de3-2530-4ee0-986e-cf0314334aec/a/bgwvQKZ#Inspect)
- **Details**: How do you plan to navigate from day to day? If this was an outline what would the subsection titles be?
   - For example:  H2 [Month/Year] February 2023; H3 [Day and Date] Monday 2; and H4 [Time period and appointment name] 8:00 a.m. Primary Care with Jonathan Schmidt
- **Image Reference**: [View an image reference to the card structure issue in the comments](https://github.com/department-of-veterans-affairs/va.gov-team/issues/57212#issuecomment-1518209388). 

##### Bandaid solution
- NA

##### Recommended solution
- **Option 1**: In order to have the subsections and the dates read out correctly by assistive technology semantic structure should be considered over using option 2 (lists). 
- **Option 2:** [Unordered lists and list items](https://design.va.gov/components/card#accessibility-considerations). Use a ul for a card group. Use an li for each card This formatting allows screen readers to enumerate the items in the card group and allows shortcuts between list items.

### Buttons & Links
- **Sketch File Location:** [Appointment List - Pre-Visit Task](https://www.sketch.com/s/0e890de3-2530-4ee0-986e-cf0314334aec/a/bgwvQKZ#Inspect)
- **Details**: The "upcoming" and "past" links should be buttons. 
- **Clarifying Questions**: Do they actually go to a new page?  Or do they change what shows on the current url and should be buttons instead? Are they acting as filters? 
- **Rationale**:  Button and link confusion can be very frustrating for assistive technology users. A user with a screen reader may pull up a list of links and may not find a specific link because it turns out that it’s been designated as a button in the markup. A good rule is if the action changes the URL, it should not be a button. [Also see: buttons and links](https://design.va.gov/components/button/#accessibility-considerations)
- **Image Reference**: [View an image reference to the buttons and links issue in the comments](https://github.com/department-of-veterans-affairs/va.gov-team/issues/57212#issuecomment-1518209388). 

Managing Focus - @SarahKay8 

![96882F43-61AD-478D-A896-E6EC23AD9B01_4_5005_c.jpeg](https://images.zenhubusercontent.com/623cd4d6774110eeb19922cb/dc9afa2d-4e9b-41a6-9275-0778ecd5a4d3)![35B161B6-FC6E-4B02-A0B1-C50389D9E006_4_5005_c.jpeg](https://images.zenhubusercontent.com/623cd4d6774110eeb19922cb/7bf6c5a8-6918-4adc-99cd-5d3095f45200)

Placement - curious on why these alerts don’t have headings?
How are we planning to manage focus, are we going to be using aria-live? It would be nice to add headings to alerts given how much content we have here. 

#### Bandaid solution
- This issue is blocked on the dependency of a design system pattern being created. 

#### Recommended solution
- If these are filters, these should be buttons. 
- For this instance, since we don’t have any design system patterns to match this, you might want to use [radio buttons](https://design.va.gov/components/form/radio-button#accessibility-considerations), to distinguish that you made a selection.

### Concise issue description
- Where you found it
- More details

#### Bandaid solution
- If helpful, provide a temporary solution that involves less content/design changes

#### Recommended solution
- Try to provide an ideal solution too that explores alternatives should content, design, IA, the broader service, etc can be fixed
- When possible, point to more research with assistive technology users

**Comment by sara-amanda** [comment]
## Non-Distinguishable Links - Image Reference

- The image displays the link "details" repeated on the screen in multiple instances. 
- ARIA labels will be needed to differentiate the links to screen readers, so that users of AT can successfully navigate the page.  
- [Additional Information on ARIA Labels](https://accessibility.18f.gov/links/)

![Screenshot 2023-04-21 at 2.37.29 PM.png](https://images.zenhubusercontent.com/63dbe59975ee0eef93e952b2/c06864ca-5d95-4883-a015-d8af37c7d9b4)

**Comment by SarahKay8** [comment]
Hey @benbrasso-agile6! Please review design feedback from @sara-amanda and I. Once reviewed, we can close this out

**Comment by benbrasso-agile6** [comment]
Thank you! We are working through some feedback we received from content. Then, will incorporate this feedback after that.

**Comment by cgednalske** [comment]
Hi @SarahKay8 👋  Following up on your recommendations -- we've addressed most of your feedback and have plans to sync with Peter Russo from the VAOS team to go over their recent a11y feedback for the appointment list, which should cover the rest. 

We're looking to schedule our midpoint review shortly and will make sure you're invited. Otherwise, unless you have additional thoughts on the changes below, I think you can go ahead and close this ticket. 🙂 

**Updates:**

- **Non-Distinguishable Links**: We have aria labels in place for these links, now documented in [an aria label section of our annotations page](https://www.sketch.com/s/0e890de3-2530-4ee0-986e-cf0314334aec/a/JAAgk0b).

- **Buttons & Links (The "upcoming" and "past" links should be buttons.)**: We've removed the upcoming and past links from our lists for now as we won't be using those for our MVP. We'll keep an eye on what happens to that component. 

- **Heading Order**: We've updated our overall heading structure ([example page demonstrating new headings](https://www.sketch.com/s/0e890de3-2530-4ee0-986e-cf0314334aec/a/bgwvQKZ)), which we believe addresses many of the heading concerns that have come up. We've also added [a spacing/heading section in our annotations](https://www.sketch.com/s/0e890de3-2530-4ee0-986e-cf0314334aec/a/oMMjpEL) to document the CSS we're using when styling headings differently than how they appear in the proper hierarchy. 

- **Card heading order**: We've adapted this from the VAOS appointment list and it sounds like they've received similar a11y feedback and are working on changes. We'll be syncing with them to address this simultaneously. 

- **Managing focus**: The design system specifies not to use headings with background-only alerts (mentioned briefly on the [alerts page](https://design.va.gov/components/alert#examples---background-color-only)), so your heading question is probably one for the design system team. I'm not sure about managing focus, but I believe we already use the alerts this way in our application, so we can definitely check with our engineers to make sure we're addressing it properly. 👍 

Thanks again for your review and let us know if there are any follow-up questions or feedback!
